lib/proxy/robust-sockjs.js

```diff
@@ -177,7 +200,80 @@ function RobustSockJSRegistry(timeout){
           robustConn.emit.apply(robustConn, arguments);
           conn._oldEmit.apply(conn, arguments);
         }
-        if (type === 'end' || type === 'close'){
+        if (type === 'data') {
+
+          // In this try block, we'll deal with the extra protocol stuff that
+          // the client adds in BufferedResendConnection. Their side can send
+          // us messages, and we can send them messages. Each side tags its
+          // messages with monotonically increasing hexadecimal message IDs
+          // (each side has its own sequence). And each side records the most
+          // recent ID it's seen.
+          //
+          // When receiving a message, there are three extra things we may need
+          // to deal with.
+          //
+          // 1) Upon disconnection and reconnection, the first message received
+          //    after reconnection MUST be "CONTINUE <id>". This tells us the
+          //    last message id from us the other side heard before we got
+          //    disconnected (actually it is <last message id> + 1). If we get
+          //    this message we must replay any missed messages. If we don't
+          //    get the CONTINUE message but a different one instead, that is
+          //    a bad connection and we should abort immediately.
+          //
+          //    When we're waiting for CONTINUE to arrive, it's not safe for
+          //    us to send any new messages; we must save them in _messageBuffer
+          //    until we know whether they've missed any messages, lest we end
+          //    up sending the messages out of order.
+          //
+          // 2) At any time (after CONTINUE) we can get an "ACK <id>" from the
+          //    other side letting us know they've received our messages up to
+          //    that id, and it's safe for us to discard them from our buffer.
+          //
+          // 3) Actual data messages are tagged with an id. We need to strip off
+          //    the id, record it, possibly send an ACK with it, and then pass
+          //    the data on.
+          try {
+            if (robustConn._expectContinue) {
+              var continueId = message_utils.parseCONTINUE(arguments[1]);
```

Comment:
Don't fail here if we see an ACK, just ignore it. It might've been triggered by an unfortunately timed event.